plugins/modules/ldap_auth_provider.py

```diff
@@ -0,0 +1,345 @@
+  sensu.sensu_go.ldap_auth_provider:
+    name: openldap
+    servers:
+      host: 127.0.0.1
+      group_search:
+        base_dn: dc=acme,dc=org
+      user_search:
+        base_dn: dc=acme,dc=org
+
+- name: Delete a LDAP auth provider
+  sensu.sensu_go.ldap_auth_provider:
+    name: openldap
+    state: absent
+"""
+
+RETURN = """
+object:
+  description: Object representing Sensu LDAP authentication provider.
+  returned: success
+  type: dict
+  sample:
+    metadata:
+      name: 'openldap'
+    servers:
+      host: '127.0.0.1'
+      port: '636'
+      insecure: 'False'
+      security: 'tls'
+      trusted_ca_file: '/path/to/trusted-certificate-authorities.pem'
+      client_cert_file: '/path/to/ssl/cert.pem'
+      client_key_file: '/path/to/ssl/key.pem'
+      binding:
+        user_dn: 'cn=binder,dc=acme,dc=org'
+        password: 'YOUR_PASSWORD'
+      group_search:
+        base_dn: 'dc=acme,dc=org'
+        attribute: 'member'
+        name_attribute': 'cn'
+        object_class: 'groupOfNames'
+      user_search:
+        base_dn: 'dc=acme,dc=org'
+        attribute: 'uid'
+        name_attribute: 'cn'
+        object_class: 'person'
+    groups_prefix: 'ldap'
+    username_prefix: 'ldap'
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors, utils
+
+API_GROUP = "enterprise"
+API_VERSION = "authentication/v2"
+
+
+def main():
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec=dict(
+            arguments.get_spec(
+                "auth",
+                "name",
+                "state",
+            ),
+            servers=dict(
+                type="list",
+                required=True,
+                elements="dict",
+                options=dict(
+                    host=dict(
+                        type="str",
+                        required=True,
+                    ),
+                    port=dict(
+                        type="int",
+                    ),
+                    insecure=dict(
+                        type="bool",
+                        default=False,
+                    ),
+                    security=dict(
+                        type="str",
+                        choices=["insecure", "tls", "starttls"],
+                        default="tls",
+                    ),
+                    trusted_ca_file=dict(
+                        type="str",
+                    ),
+                    client_cert_file=dict(
+                        type="str",
+                    ),
+                    client_key_file=dict(
+                        type="str",
+                    ),
+                    binding=dict(
+                        type="dict",
+                        options=dict(
+                            user_dn=dict(
+                                type="str",
+                            ),
+                            password=dict(
+                                type="str",
```

Comment:
We are missing `no_log=True` here because passwords should not be displayed in the output. This is also why sanity tests failed on 2.11.